### PR TITLE
fix(spaces): URL encode user names and space names (#1001)

### PR DESCRIPTION
**Pull request by stooke**
Addresses OSIO issue #1001 (Can't see any space if username contains "+")
[https://github.com/openshiftio/openshift.io/issues/1001](https://github.com/openshiftio/openshift.io/issues/1001)

There are almost certainly other components with the same problem.

The _real_ fix is to rearchitect OSIO to pass UUIDs instead of names for users, spaces, application names, etc.

### CODE REVIEW

File: src/app/spaces/space.service.spec.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ describe('Service: SpaceService', () => {
         SpaceService,
         {
           provide: WIT_API_URL,
-          useValue: "http://example.com"
+          useValue: "http://example.com/"
```
Comment (joshuawilson) [inline src/app/spaces/space.service.spec.ts]: please use single quotes

File: src/app/spaces/space.service.spec.ts [hunk 2]
```diff
@@ -132,6 +132,57 @@ describe('Service: SpaceService', () => {
...
+        'owned-by': {
+          'data': {
+            'id': '00000000-0000-0000-0000-000000000000',
+            'type': 'identities'
+          }
+        }
+      }
+    }
+  ];
+  let responseWithSlash = { data: responseDataWithSlash, links: {} };
```
Comment (joshuawilson) [inline src/app/spaces/space.service.spec.ts]: Is this variable used anywhere, I can't find it.

File: src/app/spaces/space.service.spec.ts [hunk 3]
```diff
@@ -244,6 +295,30 @@ describe('Service: SpaceService', () => {
...
+      expect(connection.request.url).toMatch(/.*test\%2Bus%20er.*/, 'url username not properly encoded');
+      expect(connection.request.url).toMatch(/.*Test\%2FSpace.*/, 'url spacename not properly encoded');
+      connection.mockRespond(new Response(
+        new ResponseOptions({
+          body: JSON.stringify({data: responseDataWithSlash[0]}),
+          status: 200
+        })
+      ));
+    });
+    let userName = "test+us er";
```
Comment (joshuawilson) [inline src/app/spaces/space.service.spec.ts]: please use single quotes

**Comment by joshuawilson** [comment]
There were more double quotes used and a few other linting errors. White space and line length.

File: src/app/spaces/space.service.spec.ts (hunk 1 above)
Comment (stooke) [reply src/app/spaces/space.service.spec.ts]: done

File: src/app/spaces/space.service.spec.ts (hunk 2 above)
Comment (stooke) [reply src/app/spaces/space.service.spec.ts]: No; deleted.

File: src/app/spaces/space.service.spec.ts (hunk 3 above)
Comment (stooke) [reply src/app/spaces/space.service.spec.ts]: Done.

**Comment by stooke** [comment]
I've fixed up most of the lint errors for the files in this commit.
(Had to learn about tslint in the first place)

**Pull request merged**